src/OpenLoco/src/Localisation/Formatting.h

```diff
@@ -0,0 +1,148 @@
+        constexpr uint8_t twoArgEnd = 22 + 1;
+
+        // Arguments (4 bytes): uint32_t
+        constexpr uint8_t inlineSpriteStr = 23;
+        // 24-31 Not used
+        constexpr uint8_t fourArgBegin = inlineSpriteStr;
+        constexpr uint8_t fourArgEnd = 31 + 1;
+
+        // Arguments in Args buffer
+        // Note:
+        // Pre formatString:
+        //     ControlCodes valid args in args buffer.
+        // Post formatString:
+        //     ControlCodes are invalid
+        //     inlineSpriteArgs replaced with inlineSpriteStr, arg is in string
+        constexpr uint8_t int32_grouped = 123 + 0;
+        constexpr uint8_t int32_ungrouped = 123 + 1;
+        constexpr uint8_t int16_decimals = 123 + 2;
+        constexpr uint8_t int32_decimals = 123 + 3;
+        constexpr uint8_t int16_grouped = 123 + 4;
+        constexpr uint8_t uint16_ungrouped = 123 + 5;
+        constexpr uint8_t currency32 = 123 + 6;
+        constexpr uint8_t currency48 = 123 + 7;
+        constexpr uint8_t stringidArgs = 123 + 8;
+        constexpr uint8_t stringidStr = 123 + 9;
+        constexpr uint8_t string_ptr = 123 + 10;
+        constexpr uint8_t date = 123 + 11;
+        constexpr uint8_t velocity = 123 + 12;
+        constexpr uint8_t pop16 = 123 + 13;
+        constexpr uint8_t push16 = 123 + 14;
+        constexpr uint8_t timeMS = 123 + 15;
+        constexpr uint8_t timeHM = 123 + 16;
+        constexpr uint8_t distance = 123 + 17;
+        constexpr uint8_t height = 123 + 18;
+        constexpr uint8_t power = 123 + 19;
+        constexpr uint8_t inlineSpriteArgs = 123 + 20;
+
+        namespace Colour
+        {
+            // Arguments: none
+            constexpr uint8_t black = 144;
+            constexpr uint8_t grey = 145;
+            constexpr uint8_t white = 146;
+            constexpr uint8_t red = 147;
+            constexpr uint8_t green = 148;
+            constexpr uint8_t yellow = 149;
+            constexpr uint8_t topaz = 150;
+            constexpr uint8_t celadon = 151;
+            constexpr uint8_t babyBlue = 152;
+            constexpr uint8_t paleLavender = 153;
+            constexpr uint8_t paleGold = 154;
+            constexpr uint8_t lightPink = 155;
+            constexpr uint8_t pearlAqua = 156;
+            constexpr uint8_t paleSilver = 157;
+        }
+    }
+
+    namespace DateModifier
+    {
+        constexpr uint8_t dmy_full = 0;
+        constexpr uint8_t my_full = 4;
+        constexpr uint8_t my_abbr = 5;
+        constexpr uint8_t raw_my_abbr = 8;
+    }
+}
+
+namespace OpenLoco
+{
+    enum class MonthId : uint8_t;
+}
+
+namespace OpenLoco::StringManager
+{
+    char* formatString(char* buffer, string_id id, const void* args = nullptr);
+    char* formatString(char* buffer, size_t bufferLen, string_id id, const void* args = nullptr);
+
+    string_id isTownName(string_id stringId);
+    string_id toTownName(string_id stringId);
+    string_id fromTownName(string_id stringId);
```

Comment:
These seem more appropriate to StringManager, to me.

Reply:
They are actually hard to categorize as this should be game specific, StringManager should only hold strings, Formatting is doing the formatting so for the time being this seemed like a somewhat reasonable place.

Reply:
I have to disagree, the plan is to move this to the Engine rather soon and in there we don't have concepts such as of what is a town, so it would need to be separated again. The fact that this function exists is a bit questionable to be honest, I haven't had the chance to look into how its used.

Reply:
Looks like these three functions are used primarily by the StationManager, fwiw.

Reply:
The primary problem is that the current design from CS is that string ids are segmented in a special way which we don't really need or have to do, during object load we just have to ask for a slot, it shouldn't matter in which slot it ends up that's why im a bit against having it as it would add a dependency where we really want a free standing string manager. I wouldn't worry too much about it, I will definitely come back to this, right now I'm trying to split up all the dependencies,  strings and string manager are currently sitting in a lot of dependency trees.

Reply:
How about we just settle for a `// TODO` note saying they'll need to be moved or refactored, then. Shouldn't be there too long.

Reply:
Added the comment, there is more than that to address in the future.